test/e2e/page-objects/pages/confirmations/redesign/gas-fee-token-modal.ts

```diff
@@ -0,0 +1,51 @@
+import { Driver } from '../../../../webdriver/driver';
+
+export default class GasFeeTokenModal {
+  constructor(driver: Driver) {
+    this.driver = driver;
+  }
+
+  protected driver: Driver;
+
+  private readonly listItem = (symbol: string) =>
+    `[data-testid="gas-fee-token-list-item-${symbol}"]`;
+
+  private readonly listItemSymbol =
+    '[data-testid="gas-fee-token-list-item-symbol"]';
+
+  private readonly listItemAmountFiat =
+    '[data-testid="gas-fee-token-list-item-amount-fiat"]';
+
+  private readonly listItemAmountToken =
+    '[data-testid="gas-fee-token-list-item-amount-token"]';
+
+  private readonly listItemBalance =
+    '[data-testid="gas-fee-token-list-item-balance"]';
+
+  async check_AmountFiat(symbol: string, amountFiat: string): Promise<void> {
+    await this.driver.findElement({
+      css: `${this.listItem(symbol)} ${this.listItemAmountFiat}`,
+      text: amountFiat,
+    });
+  }
+  async check_AmountToken(symbol: string, amountToken: string): Promise<void> {
+    await this.driver.findElement({
+      css: `${this.listItem(symbol)} ${this.listItemAmountToken}`,
+      text: amountToken,
+    });
+  }
+
+  async check_Balance(symbol: string, balance: string): Promise<void> {
+    await this.driver.findElement({
+      css: `${this.listItem(symbol)} ${this.listItemBalance}`,
+      text: balance,
+    });
+  }
+
+  async clickToken(symbol: string): Promise<void> {
+    await this.driver.clickElement({
```

Comment:
maybe we can change it to this, to prevent potential race conditions:

```suggestion
    await this.driver.clickElementAndWaitToDisappear({
```

Reply:
Done.